ambry-store/src/main/java/com.github.ambry.store/StoreMetrics.java

```diff
@@ -18,131 +18,214 @@
-    cleanupTokenFlushTime = registry.timer(MetricRegistry.name(PersistentIndex.class, name + "CleanupTokenFlushTime"));
-    hardDeleteTime = registry.timer(MetricRegistry.name(PersistentIndex.class, name + "HardDeleteTime"));
-    nonzeroMessageRecovery =
-        registry.counter(MetricRegistry.name(PersistentIndex.class, name + "NonZeroMessageRecovery"));
-    bloomPositiveCount = registry.counter(MetricRegistry.name(IndexSegment.class, name + "BloomPositiveCount"));
-    bloomFalsePositiveCount =
-        registry.counter(MetricRegistry.name(IndexSegment.class, name + "BloomFalsePositiveCount"));
-    keySizeMismatchCount = registry.counter(MetricRegistry.name(IndexSegment.class, name + "KeySizeMismatchCount"));
-    hardDeleteDoneCount = registry.counter(MetricRegistry.name(PersistentIndex.class, name + "HardDeleteDoneCount"));
-    hardDeleteFailedCount =
-        registry.counter(MetricRegistry.name(PersistentIndex.class, name + "HardDeleteFailedCount"));
-    hardDeleteIncompleteRecoveryCount =
-        registry.counter(MetricRegistry.name(PersistentIndex.class, name + "HardDeleteIncompleteRecoveryCount"));
-    hardDeleteExceptionsCount =
-        registry.counter(MetricRegistry.name(PersistentIndex.class, name + "HardDeleteExceptionsCount"));
-    segmentSizeForExists = registry.histogram(MetricRegistry.name(IndexSegment.class, name + "SegmentSizeForExists"));
   }
 
-  public void initializeCapacityUsedMetric(final Log log, final long capacityInBytes) {
-    currentCapacityUsed = new Gauge<Long>() {
-      @Override
-      public Long getValue() {
-        return log.getLogEndOffset();
-      }
-    };
-    registry.register(MetricRegistry.name(Log.class, name + "CurrentCapacityUsed"), currentCapacityUsed);
-    percentageUsedCapacity = new Gauge<Double>() {
-      @Override
-      public Double getValue() {
-        return ((double) log.getLogEndOffset() / capacityInBytes) * 100;
-      }
-    };
-    registry.register(MetricRegistry.name(Log.class, name + "PercentageUsedCapacity"), percentageUsedCapacity);
+  /**
+   * Create a {@link DiskLevelMetrics} object for handling metrics related to the stores on a certain disk and make it
+   * accessible for lookup.
+   * @param diskId the name of the disk.
+   * @return the {@link DiskLevelMetrics}
+   */
+  public DiskLevelMetrics createDiskLevelMetrics(String diskId) {
+    DiskLevelMetrics metrics = new DiskLevelMetrics(diskId, registry);
+    diskLevelMetrics.put(diskId, metrics);
+    return metrics;
+  }
+
+  /**
+   * @param diskId the name of the disk to lookup the metrics object for.
+   * @return the {@link DiskLevelMetrics} for a disk name, or {@code null} if that disk could not be found.
+   */
+  public DiskLevelMetrics getDiskLevelMetrics(String diskId) {
+    return diskLevelMetrics.get(diskId);
+  }
+
+  /**
+   * Metrics for a specific disk.
+   */
+  public static class DiskLevelMetrics {
```

Comment:
Good abstraction. Now, its easy for us to track any metric on a disk level :) We should try to shortlist those metrics which makes sense a disk, and add them at disk level. May be in next PR